src/rules/flake8_use_pathlib/helpers.rs

```diff
@@ -0,0 +1,238 @@
+                        None
+                    }
+                }
+                ["os", "path", "join"] => {
+                    if checker.settings.rules.enabled(&Rule::PathlibJoin) {
+                        Some(OsCall::Join)
+                    } else {
+                        None
+                    }
+                }
+                ["os", "path", "basename"] => {
+                    if checker.settings.rules.enabled(&Rule::PathlibBasename) {
+                        Some(OsCall::Basename)
+                    } else {
+                        None
+                    }
+                }
+                ["os", "path", "dirname"] => {
+                    if checker.settings.rules.enabled(&Rule::PathlibDirname) {
+                        Some(OsCall::Dirname)
+                    } else {
+                        None
+                    }
+                }
+                ["os", "path", "samefile"] => {
+                    if checker.settings.rules.enabled(&Rule::PathlibSamefile) {
+                        Some(OsCall::Samefile)
+                    } else {
+                        None
+                    }
+                }
+                ["os", "path", "splitext"] => {
+                    if checker.settings.rules.enabled(&Rule::PathlibSplitext) {
+                        Some(OsCall::Splitext)
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            })
+    {
+        let diagnostic = Diagnostic::new::<DiagnosticKind>(
+            match os_call {
+                OsCall::Abspath => PathlibAbspath.into(),
+                OsCall::Chmod => PathlibChmod.into(),
+                OsCall::Mkdir => PathlibMkdir.into(),
+                OsCall::Makedirs => PathlibMakedirs.into(),
+                OsCall::Rename => PathlibRename.into(),
+                OsCall::Replace => PathlibReplace.into(),
+                OsCall::Rmdir => PathlibRmdir.into(),
+                OsCall::Remove => PathlibRemove.into(),
+                OsCall::Unlink => PathlibUnlink.into(),
+                OsCall::Getcwd => PathlibGetcwd.into(),
+                OsCall::Exists => PathlibExists.into(),
+                OsCall::Expanduser => PathlibExpanduser.into(),
+                OsCall::IsDir => PathlibIsDir.into(),
+                OsCall::IsFile => PathlibIsFile.into(),
+                OsCall::IsLink => PathlibIsLink.into(),
+                OsCall::Readlink => PathlibReadlink.into(),
+                OsCall::Stat => PathlibStat.into(),
+                OsCall::IsAbs => PathlibIsAbs.into(),
+                OsCall::Join => PathlibJoin.into(),
+                OsCall::Basename => PathlibBasename.into(),
+                OsCall::Dirname => PathlibDirname.into(),
+                OsCall::Samefile => PathlibSamefile.into(),
+                OsCall::Splitext => PathlibSplitext.into(),
+            },
+            Range::from_located(expr),
+        );
```

Comment:
Is it possible to get rid of all the `if checker.settings.rules.enabled` in the above `map` call, then here, create the `Diagnostic`, and do `settings.rules.enabled(diagnostic.kind.rule())` prior to pushing?

Reply:
Makes sense, pushed the changes. We already checked that any of them is enabled. We might even be able to get rid of `OsCall` altogether

Reply:
Yeah you could probably get rid of `OsCall` here?

Reply:
Fixed in https://github.com/charliermarsh/ruff/pull/2091